Provider/src/main/java/com/spotify/confidence/cache/StorageFileCache.kt

```diff
@@ -1,18 +1,26 @@
 package com.spotify.confidence.cache
 
 import android.content.Context
+import com.spotify.confidence.apply.ApplyInstance
 import com.spotify.confidence.client.ResolvedFlag
+import com.spotify.confidence.client.serializers.UUIDSerializer
+import dev.openfeature.sdk.DateSerializer
 import dev.openfeature.sdk.EvaluationContext
 import dev.openfeature.sdk.Value
 import kotlinx.serialization.Serializable
 import kotlinx.serialization.encodeToString
 import kotlinx.serialization.json.Json
+import kotlinx.serialization.modules.SerializersModule
+import kotlinx.serialization.modules.contextual
 import java.io.File
 
 const val FLAGS_FILE_NAME = "confidence_flags_cache.json"
+const val APPLY_FILE_NAME = "confidence_apply_cache.json"
 
-class StorageFileCache private constructor(context: Context) : DiskStorage {
-    private val file: File = File(context.filesDir, FLAGS_FILE_NAME)
+class StorageFileCache private constructor(
```

Comment:
great stuff! can we go one step further and make this private/internal?

Reply:
Good point. fixed!

Reply:
We could probably go through and do this for more classes, maybe not the interfaces but definitely the implementations.